Added highlighting to search feature

ajleon95 opened the pull request:
1. The container div for the content now has id `main` to simplify the JavaScript. 
2. The search page now appends GET parameter `q` to each result link to allow for highlighting. 
3. The search form no longer submits upon hitting enter to prevent the user from accidentally losing their query and results. 
4. A small JavaScript script gets the parameter `q`, processes the result, and adds the tag `<mark>` around any matches. 
5. `color` property on `mark` was set to `inherit` to match the rest of Ed's theme (defaults to black). 

As of right now the script will not highlight a term that is split by a footnote (e.g. searching "bugle trills" will return "O Captain! My Captain!" but will not be highlighted), but will work *within* footnotes. Now with fixed variables it will work more reliably (and highlight the static text on the search page without breaking functionality, if you want).

elotroalex commented:
Armando, I noticed just now that you had submitted a pull-request. For some reason I didn't get the usual email. I went ahead and implemented the search highlighting myself based on your ideas before seeing the pull request. Now we have conflicts. I would like you to get credit for this work both in the credit page, but also in the commit history. Can you take a look at what the differences are between what I did and this? We can also simply talk on the phone soon.

ajleon95 commented:
Hi Alex, looks like the conflict probably comes from search.js and main.css. I changed the implementation for Ed from the one we used on minilazarillo. On Ed I simply stopped the browser from submitting the form when pressing enter. This means you can't share the search results with a URL, but you won't lose your query if you happen to press enter out of habit. A consequence of this is that you do not need to separate out the function to `doSearch()`. I also changed the way you get parameters and information from the URL. Since we only have one parameter `q` I used a substring. Both changes should make it more obvious what is happening in search.js. 

For main.css it's just a question of style, so I defer to you there. I'm not sure how to resolve the conflict (you might have to do it) but I think the code I've used here for search.js would be more obvious and clearer to understand.

elotroalex commented:
Ok. I can work with this. Let me do some diff analysis to see how to resolve the merge

elotroalex commented:
Armando, I fixed this more or less with your code via another commit. Thanks for such a wonderful feature!
